apps/common/serializers.py

```diff
@@ -58,3 +59,36 @@ def update(self, instance: ModelType, validated_data: dict[str, typing.Any]) ->
             validated_data["modified_by"] = self.context["request"].user
 
         return super().update(instance, validated_data)
+
+
+class ArchivableResourceSerializer[ModelType: ArchivableResource, ContextType: DrfContextType = DrfContextType](
+    serializers.ModelSerializer[ModelType],
+):
+    instance: ModelType | None  # type: ignore[override]
+
+    @property
+    def context(self) -> ContextType:  # type: ignore[override]
+        ctx = super().context
+        assert ctx is not None, f"Always pass context when using {type(self)}"
+        return typing.cast("ContextType", ctx)
+
+    def _set_archive_fields(self, validated_data: dict[str, typing.Any], instance: ModelType) -> None:
```

Comment:
We need to add "is_archived" field here so that it does not need to be repeated in each class where it's used